src/data_structures/hashtable.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::LinkedList;
+
+pub struct HashTable<K, V> {
+    elements: Vec<LinkedList<(K, V)>>,
+    count: usize,
+}
+
+impl<K: Hashable + std::cmp::PartialEq, V> Default for HashTable<K, V> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub trait Hashable {
+    fn hash(&self) -> usize;
+}
+
+impl<K: Hashable + std::cmp::PartialEq, V> HashTable<K, V> {
+    pub fn new() -> HashTable<K, V> {
+        let initial_capacity = 3000;
+        let mut elements = Vec::with_capacity(initial_capacity);
+
+        for _ in 0..initial_capacity {
+            elements.push(LinkedList::new());
+        }
+
+        HashTable { elements, count: 0 }
+    }
+
+    pub fn insert(&mut self, key: K, value: V) {
+        if self.count >= self.elements.len() * 3 / 4 {
```

Comment:
nit: Extract to a constant. `const LOAD_FACTOR_BOUND: f32 = 0.75;`.

This will make this section a bit uglier because of the casts, but the constant will be explained. wdyt?